### PR TITLE
migrate to TWGC

### PR DESCRIPTION
**Pull request by davidkhala**
migrate from domain `guoger` to TWGC
Signed-off-by: DavidLiu <david.yx.liu@oracle.com>

### CODE REVIEW

**Comment by SamYuan1990** [comment]
seems missing for docker image push and where is the docker repo url?

**Comment by guoger** [comment]
> seems missing for docker image push and where is the docker repo url?

PR looks good to me. Currently my dockerhub is configured with a CD pipeline that *watches this repo* and automatically build image. It seems pretty robust and simple. Not sure if the same could be done with new docker registry?

**Comment by SamYuan1990** [comment]
> > seems missing for docker image push and where is the docker repo url?
> 
> PR looks good to me. Currently my dockerhub is configured with a CD pipeline that **watches this repo** and automatically build image. It seems pretty robust and simple. Not sure if the same could be done with new docker registry?

yep... 1st of all, I want to know how to access TWGC docker repo. then to see if further changes need to be done in this pr as azp pipeline part, making merge into master/main trigger a latest image build. etc.

**Comment by davidkhala** [comment]
@SamYuan1990 
image
This is where you can find the published package. 
`ghcr.io/hyperledger-twgc/tape` is the docker repository URL. 

I have not yet established a pipeline to continuous-publish. But something I wonder is according to Github policy, maintainer of this repository also have right to publish docker image to organization level.

**Comment by davidkhala** [comment]
> CD

I believe it is easier to use Github Action to do so. I would have a try.

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: release latest
+
+on:
+  push:
+    branches: ['master']
```
Comment (SamYuan1990) [inline .github/release.yml:5]: add alpha & v.*(release branch) please.

**Review by SamYuan1990 — CHANGES_REQUESTED** [review]
or @davidkhala you want to have another commit for alpha and release branch?

File: .github/release.yml (hunk 1 above)
Comment (davidkhala) [reply .github/release.yml:5]: alpha或者release branch发布出来的docker image tag应该怎么打？打成...tape:alpha 以及tape:1.1.1这样么？

**Comment by davidkhala** [comment]
> > @SamYuan1990
> > image
> > This is where you can find the published package.
> > `ghcr.io/hyperledger-twgc/tape` is the docker repository URL.
> > I have not yet established a pipeline to continuous-publish. But something I wonder is according to Github policy, maintainer of this repository also have right to publish docker image to organization level.
> 
> I can't find it. pls help with my permission.

这个不需要权限的，已经设为全网公开的，是docker pull出的问题还是网页上看不到了？

File: .github/release.yml (hunk 1 above)
Comment (SamYuan1990) [reply .github/release.yml:5]: tape:alpha 以及tape:1.1.1

**Comment by SamYuan1990** [comment]
> > > @SamYuan1990
> > > image
> > > This is where you can find the published package.
> > > `ghcr.io/hyperledger-twgc/tape` is the docker repository URL.
> > > I have not yet established a pipeline to continuous-publish. But something I wonder is according to Github policy, maintainer of this repository also have right to publish docker image to organization level.
> > 
> > 
> > I can't find it. pls help with my permission.
> 
> 这个不需要权限的，已经设为全网公开的，是docker pull出的问题还是网页上看不到了？

```
yuanyi@192 ~ % docker pull ghcr.io/hyperledger-twgc/tape
Using default tag: latest
latest: Pulling from hyperledger-twgc/tape
29291e31a76a: Already exists 
f10fb7ea8362: Pull complete 
0b219d765d96: Pull complete 
Digest: sha256:c859af8ce24cbd94e58ba20b2a5ec58b7a644bd58735f8b837845ddf72d00f37
Status: Downloaded newer image for ghcr.io/hyperledger-twgc/tape:latest
ghcr.io/hyperledger-twgc/tape:latest
```

**Comment by SamYuan1990** [comment]
> 现在这个改法出来的效果是
> 如果push to master branch，会出tape:master, 而不是tape:latest
> 如果push 一个tag如v1.3.2，则会同时出tape:latest和tape:v1.3.2
> 
> 转化关系见https://github.com/docker/metadata-action#basic 锚点下方的语法关系表

我们试试 这个？ https://github.com/Hyperledger-TWGC/tape/pull/198#issuecomment-897777133

或者 这个 https://github.com/docker/metadata-action#latest-tag

我觉得不能随意调整tag的话，对于我们有些难受了。
实际上我们通过latest发布main/master上的最新代码，通过alpha来发布一些最新改动，通过v.*的tag来做发布版本。

**Comment by davidkhala** [comment]
> > 现在这个改法出来的效果是
> > 如果push to master branch，会出tape:master, 而不是tape:latest
> > 如果push 一个tag如v1.3.2，则会同时出tape:latest和tape:v1.3.2
> > 转化关系见https://github.com/docker/metadata-action#basic 锚点下方的语法关系表
> 
> 我们试试 这个？ https://github.com/docker/metadata-action#typeedge
> 
> 或者 这个 https://github.com/docker/metadata-action#latest-tag
> 
> 我觉得不能随意调整tag的话，对于我们有些难受了。
> 实际上我们通过latest发布main/master上的最新代码，通过alpha来发布一些最新改动，通过v.*的tag来做发布版本。

我对当前版本这样的理解对么？
- 对v.*的git tag的发布是OK的
- master发latest, alpha branch 发edge

**Comment by SamYuan1990** [comment]
> * master发latest, alpha branch 发edge

是的。如果能做到这样是最好的。

**Comment by davidkhala** [comment]
> > * master发latest, alpha branch 发edge
> 
> 是的。如果能做到这样是最好的。

现在这个版本应该能做到这个效果，不过要merge之后才能看到效果

**Review by SamYuan1990 — APPROVED** [review]

**Pull request merged**